### DISCUSSION TITLE
MuSHR for middle-school kids?

### DISCUSSION THREAD

**Discussion by rttp224** [prl-mushr/mushr#22] [Q&A] [answered]
Hi,

I am trying to mentor some neighborhood middle-school kids to learn about robotics and autonomous driving in a fun environment. I am planning to have the kids assemble the hardware and deploy the software using the tutorials on MuSHR's website. Two questions:

1) Are there other groups (with middle-school children) able to build the car successfully?

2) Assuming the kids can spend 2 hrs/day for 3 days/week, is this something doable over 2 months? 

Thanks

**Comment by zlite** [comment, marked as answer]
My two cents: this is WAY too hard (and expensive) for middle schoolers.  Try something under $100 that can be built in a day, like this: https://diyrobocars.com/2017/10/01/a-minimum-viable-racer-for-openmv/

**Reply by rttp224** [reply]
Well... I didn't provide the complete context. The middle schoolers I am talking about are actually 13-year olds (completing their 7th grade). They are pretty proficient with computers, they have assembled a computer by themselves ordering parts from Amazon. They have some basic coding experience in javascript/node.js .

The $100 OpenMV looks great, but I am afraid it can be too simple for them, they might be able to finish it off in a couple of days. I was trying to see if there is some project that is a bit more challenging for them. I am proficient with Linux, Python etc. So I thought I could give them some guidance on Linux, Python and let the teenagers go wild on the incredibly well-written tutorials. 

If MuSHR is too hard for 13-year olds, are there other projects you would recommend that are at intermediate-level?

**Reply by zlite** [reply]
OpenMV is the complete OpenCV stack. If your 13 year olds are too advanced for that, they should skip high school entirely and go straight to a CS major at a top university ;-)

Have you done a full semester of CV with middle schoolers before? It's challenging. It was a PhD about three years ago.

**Reply by schmittlema** [reply]
Hey @rttp224! Chris brings up a good point. The MuSHR cars were not designed for middle school students but rather late high school --- grad school. We've had a juniors/seniors work with the cars before. The main difficulty comes from linux. If students aren't particularly familiar with linux, then using ROS can be challenging, and debugging becomes impossible. To others wanting to get their middle schoolers engaged in robotics we usually recommend starting with something based around Rasberry Pi like the [Donkey Car](https://www.donkeycar.com/). Rasberry Pi alone has lots of tutorials and resources that can help them build the foundational knowledge of linux and python (thats how I started!). @zlite would know more about Donkey Car and if that is the right fit, but generally I recommend starting with simple robots based around Rasberry Pi.

Here is a couple other Rasberry Pi robots of various complexities I found:

- https://projects.raspberrypi.org/en/projects/build-a-buggy
- https://www.amazon.com/Learn-Robotics-Raspberry-Pi-Thinking/dp/1593279205
- https://www.piborg.org/robots-1/monsterborg <---- looks awesome!
- https://www.robotshop.com/en/sunfounder-picar-s-raspberry-pi-4-32b-smart-car-kit.html?gclid=CjwKCAjwnPOEBhA0EiwA609RedXi-7uRwFS6AAi017VzRzkkmuLtCuT1sviUCrpBv2c6ajubWxBNPhoCvXwQAvD_BwE
- https://www.seeedstudio.com/AlphaBot-Raspberry-Pi-robot-building-kit-p-2842.html?utm_source=blog&utm_medium=blog

**Reply by rttp224** [reply]
> OpenMV is the complete OpenCV stack. If your 13 year olds are too advanced for that, they should skip high school entirely and go straight to a CS major at a top university ;-)
> 
> Have you done a full semester of CV with middle schoolers before? It's challenging. It was a PhD about three years ago.

Well... The 13-year olds I am thinking of are not that advanced to skip high-school :) 

I worked for a CV company before, so I am aware of the complexity of CV algorithms. I wasn't intending to have the teenagers go through the theory as they barely touched basic algebra, let alone linear algebra or calculus. My intention was for them to use CV from a black-box approach and get excited with the possibility. 

I do understand your perspective though...

**Comment by rttp224** [comment]
Hi @schmittlema <https://github.com/schmittlema>: Thank you very much for the feedback on MuSHR and the detailed list of potential other robot projects. The Donkey Car and MonsterBorg both look promising! 

Thanks again!
